### PR TITLE
feat(dpp): [v23 port] Identity Update Transition

### PR DESCRIPTION
**Pull request by qrayven**
## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added or updated relevant unit/integration/functional/e2e tests
- [x] I have made corresponding changes to the documentation

**For repository code-owners and collaborators only**
- [ ] I have assigned this pull request to a milestone

### CODE REVIEW

File: dpp/src/identity/identity_public_key.rs [hunk 1]
```diff
@@ -178,6 +179,9 @@ pub struct IdentityPublicKey {
     pub key_type: KeyType,
     pub data: Vec<u8>,
     pub read_only: bool,
+
+    #[serde(skip_serializing_if = "Option::is_none")]
```
Comment (antouhou) [inline dpp/src/identity/identity_public_key.rs:183]: Oh, didn't know that there's a macro for that in serde, neat!

File: dpp/src/tests/identity/state_transition/identity_update_transition/identity_update_transition_spec.rs [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+fn to_object_with_signature_skipped() {
+    let TestData { transition, .. } = setup_test();
+    let result = transition
+        .to_object(true)
+        .expect("conversion to object shouldn't fail");
+
+    let expected_raw_state_transition = json!({
+        "protocolVersion" : 1,
+        "type" : 5,
+        "signaturePublicKeyId": 0,
```
Comment (antouhou) [inline dpp/src/tests/identity/state_transition/identity_update_transition/identity_update_transition_spec.rs:170]: Should the id still be there if we've removed the signature? I've actually never thought about this before - in JS DPP it seems we do the same thing. Probably something to discuss at the architecture meeting, @shumkov @jawid-h @shuplenkov

Comment (antouhou) [reply dpp/src/tests/identity/state_transition/identity_update_transition/identity_update_transition_spec.rs:170]: I guess the logic was that when you call the `.sign` method in JS this field is unset. Not sure

Comment (antouhou) [reply dpp/src/tests/identity/state_transition/identity_update_transition/identity_update_transition_spec.rs:170]: (This comment shouldn't prevent the PR from being merged, as it is in line with the original implementation)

File: dpp/src/tests/identity/validation/required_purpose_and_security_level_validator_spec.rs [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+
+    assert!(matches!(
+        result.errors()[0],
+        crate::consensus::ConsensusError::MissingMasterPublicKeyError(..)
+    ));
+    assert_eq!(1046, result.errors()[0].code())
+}
+
+#[test]
+fn should_return_valid_resutl() {
```
Comment (antouhou) [inline dpp/src/tests/identity/validation/required_purpose_and_security_level_validator_spec.rs]: `result`

File: dpp/src/tests/utils/error_helpers.rs [hunk 4]
```diff
@@ -33,6 +34,23 @@ pub fn get_index_error(consensus_error: &ConsensusError) -> &IndexError {
...
+    result: &ValidationResult<()>,
+    error_number: usize,
+) -> &StateError {
+    match result
+        .errors
+        .get(error_number)
+        .expect("error should be found")
+    {
+        ConsensusError::StateError(state_error) => state_error,
+        _ => panic!(
```
Comment (antouhou) [inline dpp/src/tests/utils/error_helpers.rs:47]: Do we really want that to panic or just return `None`?

Comment (QuantumExplorer) [reply dpp/src/tests/utils/error_helpers.rs:47]: Meh, maybe change this later.

**Review by QuantumExplorer — APPROVED** [review]

**Pull request merged**